orttraining/orttraining/core/framework/mpi_setup.h

```diff
@@ -1,6 +1,8 @@
 #pragma once
 
-#if defined(USE_NCCL) || defined(USE_HOROVOD)
+#if defined(USE_MPI)
+// Disable deprecated C++ bindings, which fail to type-check with GCC 9.3
+#define OMPI_SKIP_MPICXX
 #include <mpi.h>
```

Comment:
nccl_common.cc uses mpi to initialize nccl context and it's not guarded by anything at the moment. Maybe we should add the check there as well.